bas-platform/src/actions/impl.ts

```diff
@@ -33,19 +34,19 @@ export class CommandAction extends Action implements ICommandAction {
 export class SnippetAction extends Action implements ISnippetAction {
     contributorId: string;
     snippetName: string;
-    context: any;
+    context: SnippetActionParams;
 
     constructor() {
         super();
         this.actionType = ActionType.Snippet;
         this.contributorId = "";
-        this.context = "";
         this.snippetName = "";
+        this.context = "";
     }
 }
 
 export class FileAction extends Action implements IFileAction {
-    uri: Uri;
+    uri: FileActionParams;
```

Comment:
It seems that `uri: vscode.Uri` is much clearer. I'm not sure we need to make changes to `bas-platform` at all...

Reply:
But then what if bas-platform api will chabge then we'll have to change guided-dev api.

Reply:
> But then what if bas-platform api will chabge then we'll have to change guided-dev api.

True. I'll approve the PR. Just remove the sample code from `types.ts`